cardano-node/src/Cardano/Tracing/Startup.hs

```diff
@@ -0,0 +1,244 @@
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+module Cardano.Tracing.Startup where
+
+import           Prelude
+
+import           Data.Aeson (Value (..), (.=))
+import qualified Data.Aeson as Aeson
+import           Data.List (intercalate)
+import           Data.Map (Map)
+import qualified Data.Map as Map
+import           Data.Text (Text)
+import qualified Data.Text as Text
+import           Data.Time.Clock (UTCTime)
+import           Data.Time.Clock.POSIX (POSIXTime, utcTimeToPOSIXSeconds)
+
+import           Cardano.Slotting.Slot (SlotNo (..))
+import           Cardano.Node.Configuration.TopologyP2P (UseLedger (..))
+import           Cardano.Node.Configuration.Socket
+import           Cardano.Tracing.OrphanInstances.Network ()
+
+import           Cardano.BM.Tracing (HasPrivacyAnnotation (..),
+                   HasSeverityAnnotation (..), Severity (..), ToObject (..),
+                   Transformable (..))
+import           Cardano.BM.Data.Tracer (HasTextFormatter (..), mkObject,
+                   trStructuredText)
+
+import           Ouroboros.Consensus.Node.NetworkProtocolVersion
+                   (BlockNodeToClientVersion, BlockNodeToNodeVersion)
+
+import           Ouroboros.Network.Magic (NetworkMagic (..))
+import           Ouroboros.Network.NodeToClient (LocalAddress (..),
+                   LocalSocket, NodeToClientVersion)
+import           Ouroboros.Network.NodeToNode (DiffusionMode (..),
+                   NodeToNodeVersion)
+import           Ouroboros.Network.PeerSelection.LedgerPeers (UseLedgerAfter (..))
+import           Ouroboros.Network.PeerSelection.Types (PeerAdvertise)
+import           Ouroboros.Network.PeerSelection.RelayAccessPoint (RelayAccessPoint)
+
+import qualified Network.Socket as Socket
+
+
+data StartupTrace blk =
+    StartupInfo
+      [SocketOrSocketInfo Socket.SockAddr Socket.SockAddr]
+      (Maybe (SocketOrSocketInfo LocalSocket LocalAddress))
+      (Map NodeToNodeVersion (BlockNodeToNodeVersion blk))
+      (Map NodeToClientVersion (BlockNodeToClientVersion blk))
+
+  | StartupP2PInfo DiffusionMode
+    
+  | StartupTime UTCTime
+
+  | StartupNetworkMagic NetworkMagic
+
+  | StartupSocketConfigError SocketConfigError
+
+  | StartupDBValidation
+
+  | NetworkConfigUpdate
+
+  | NetworkConfigUpdateError Text
+
+  | NetworkConfig [(Int, Map RelayAccessPoint PeerAdvertise)]
+                  [RelayAccessPoint]
+                  UseLedgerAfter
+
+  | P2PWarning
+
+  | P2PWarningDevelopementNetworkProtocols
```

Comment:
We should drop the P2P from this message. Currently `DevelopementNetworkProtocols` means p2p protocols, but that will not be the case in the future.

Reply:
This is a specific warning for p2p: it requires `TestEnableDevelopmentNetworkProtocols`, so when it is not set it will show a message that one needs to enable it.  But I take your point that we should have another warning that says that development network protocols are in use.

Reply:
Done, it's either:
```
[waldorf:cardano.node.nodeconfig:Warning:5] [2021-11-23 11:50:47.86 UTC] unsupported and unverified version of `cardano-node` with peer-to-peer networking capabilities
[waldorf:cardano.node.nodeconfig:Warning:5] [2021-11-23 11:50:47.86 UTC] using development network protocols: [NodeToNodeV_8] []
```
or
```
[waldorf:cardano.node.nodeconfig:Warning:5] [2021-11-23 11:53:36.04 UTC] unsupported and unverified version of `cardano-node` with peer-to-peer networking capabilities
[waldorf:cardano.node.nodeconfig:Warning:5] [2021-11-23 11:53:36.04 UTC] peer-to-peer requires TestEnableDevelopmentNetworkProtocols to be set to True
```
or nothing, or just the development network protocols message.
```